server/app/services/ProductionService.ts

```diff
@@ -0,0 +1,100 @@
+import TYPES from '../constants/types';
+import { IProduction } from './../models/ProductionModel';
+import { inject, injectable } from 'inversify';
+import { ProductionRepository } from './../repository/ProductionRepository';
+import { BadRequestError, NotFoundError } from '../errors';
+import { BikeService } from './BikeService';
+import { PartService } from './PartService';
+import PartEntity from './../entities/Part';
+import BikeEntity from './../entities/Bike';
+import { IBike } from './../models/BikeModel';
+import { IPart } from './../models/PartModel';
+
+@injectable()
+export class ProductionService {
+  constructor(
+    @inject(TYPES.ProductionRepository) private productionRepository: ProductionRepository,
+    @inject(TYPES.BikeService) private bikeService: BikeService,
+    @inject(TYPES.PartService) private partService: PartService
+  ) {}
+
+  /**
+   * Return all productions json objects
+   * @returns all productions
+   */
+  public async getProductions(): Promise<IProduction[]> {
+    return this.productionRepository.getList();
+  }
+
+  /**
+   * Create a new production json obj
+   * @param body production json obj
+   * @returns production json obj
+   */
+  public async createProduction(body: IProduction): Promise<IProduction> {
+    if (body.type === 'Part') {
+      const partEntity = {
+        name: body.componentDetail.name,
+        quality: body.componentDetail.quality,
+        description: body.componentDetail.description,
+        type: body.componentDetail.type,
+        color: body.componentDetail.color,
+        finish: body.componentDetail.finish,
+        grade: body.componentDetail.grade,
+        detail: body.componentDetail.detail,
+        stock: body.quantity,
+        profitMargin: body.componentDetail.profitMargin,
+      } as IPart;
+      const validPart = await PartEntity.validate(partEntity, 'create');
+      const partCreated = await this.partService.createPart(validPart);
+      body.componentId = partCreated._id;
+      return this.productionRepository.create(body);
+    }
+
+    if (body.type === 'Bike') {
+      const bikeEntity = {
+        name: body.componentDetail.name,
+        description: body.componentDetail.description,
+        weightAmount: body.componentDetail.weightAmount,
+        weightType: body.componentDetail.weightType,
+        color: body.componentDetail.color,
+        parts: body.componentDetail.parts,
+        stock: body.quantity,
+        profitMargin: body.componentDetail.profitMargin,
+      } as IBike;
+      const validBike = await BikeEntity.validate(bikeEntity, 'create');
```

Comment:
```suggestion
      const validBike = await BikeEntity.validate(
        ({ ...body.componentDetail, stock: body.quantity } as unknown) as IBike,
        'create'
      );
```

This might be cleaner? (Can do similarly with `IPart` above)

@Mackbellemore Is it equivalent?

Reply:
As long as componentDetail doesnt have extra stuff that a Bike doesnt have then ya something like this works.

Reply:
@gordonpn NICE! it's much cleaner I agree. 👍